### DISCUSSION TITLE
Trying to Get My Localhost Going

### DISCUSSION THREAD

**Discussion by websitejunction** [Homebrew/discussions#4320] [Getting started]
### Output of `brew config`

```shell
willbermudez@WIlls-Mac-mini / % brew config
HOMEBREW_VERSION: 4.0.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: 0c1031f2fd71016f0f5df699caaed384487bfb16
Last commit: 2 days ago
Core tap JSON: 12 Mar 04:30 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
willbermudez@WIlls-Mac-mini / % brew doctor
==> Downloading https://formulae.brew.sh/api/formula.jws.json

Your system is ready to brew.
```

### Description of issue

I have been going in circles for the last two days now. I really hope someone can answer or help me get going here. 

After installing homebrew, I installed the following:

```
Name    Status     User         File
emacs   started    willbermudez ~/Library/LaunchAgents/homebrew.mxcl.emacs.plist
httpd   error  256 root         ~/Library/LaunchAgents/homebrew.mxcl.httpd.plist
mysql   stopped    root         ~/Library/LaunchAgents/homebrew.mxcl.mysql.plist
php     started    willbermudez ~/Library/LaunchAgents/homebrew.mxcl.php.plist
unbound error  256 willbermudez ~/Library/LaunchAgents/homebrew.mxcl.unbound.plist
```

When attempting: brew services start mysql

```
Bootstrap failed: 5: Input/output error
Try re-running the command as root for richer errors.
Error: Failure while executing; `/bin/launchctl bootstrap gui/501 /Users/willbermudez/Library/LaunchAgents/homebrew.mxcl.mysql.plist` exited with 5.
```

I tried: brew cleanup --prune=all

Then: sudo brew services start mysql

```
Warning: Taking root:admin ownership of some mysql paths:
  /opt/homebrew/Cellar/mysql/8.0.32/bin
  /opt/homebrew/Cellar/mysql/8.0.32/bin/mysqld_safe
  /opt/homebrew/opt/mysql
  /opt/homebrew/opt/mysql/bin
  /opt/homebrew/var/homebrew/linked/mysql
This will require manual removal of these paths using `sudo rm` on
brew upgrade/reinstall/uninstall.
Warning: mysql must be run as non-root to start at user login!
/Library/LaunchDaemons/homebrew.mxcl.mysql.plist: service already bootstrapped
Bootstrap failed: 37: Operation already in progress
Error: Failure while executing; `/bin/launchctl bootstrap system /Library/LaunchDaemons/homebrew.mxcl.mysql.plist` exited with 37.
```
sudo brew services restart httpd

```
Stopping `httpd`... (might take a while)
==> Successfully stopped `httpd` (label: homebrew.mxcl.httpd)
Warning: Taking root:admin ownership of some httpd paths:
  /opt/homebrew/Cellar/httpd/2.4.56/bin
  /opt/homebrew/Cellar/httpd/2.4.56/bin/httpd
  /opt/homebrew/opt/httpd
  /opt/homebrew/opt/httpd/bin
  /opt/homebrew/var/homebrew/linked/httpd
This will require manual removal of these paths using `sudo rm` on
brew upgrade/reinstall/uninstall.
Warning: httpd must be run as non-root to start at user login!
==> Successfully started `httpd` (label: homebrew.mxcl.httpd)
willbermudez@WIlls-Mac-mini / % sudo ps -A |grep httpd
66736 ??         0:00.29 /opt/homebrew/opt/httpd/bin/httpd -k start
66737 ??         0:00.00 /opt/homebrew/opt/httpd/bin/httpd -k start
66738 ??         0:00.07 /opt/homebrew/opt/httpd/bin/httpd -k start
66739 ??         0:00.01 /opt/homebrew/opt/httpd/bin/httpd -k start
66740 ??         0:00.00 /opt/homebrew/opt/httpd/bin/httpd -k start
66741 ??         0:00.01 /opt/homebrew/opt/httpd/bin/httpd -k start
68355 ??         0:00.01 /opt/homebrew/opt/httpd/bin/httpd -k start
73276 ??         0:00.00 /opt/homebrew/opt/httpd/bin/httpd -k start
 1834 ttys002    0:00.00 grep httpd
willbermudez@WIlls-Mac-mini / % sudo apachectl start
httpd (pid 66736) already running

```
I can access mysql with a password I was able to create using:

mysql -uroot -p -h127.0.0.1 --protocol=tcp

There is nothing in: /user/local/var/mysql

I've been on chatGPT and there was some help but I'm going in circles with it now.

Any insight is GREATLY appreciated.

**Comment by gromgit** [comment]
>```
>mysql   stopped    root         ~/Library/LaunchAgents/homebrew.mxcl.mysql.plist
>```
The MySQL folks specifically recommend _NOT_ running as root (https://mariadb.com/kb/en/running-mysqld-as-root/), and nothing in the Homebrew setup for `mysql` requires it. Homebrew also gave you another reason not to do so:
>```
>Warning: mysql must be run as non-root to start at user login!
>```
Also:
>```sh
>% sudo ps -A |grep httpd
>```
You don't need `sudo` here, but you _do_ need to add the `-f` flag to `ps`, so that it shows you the user actually running the process (this is critical info).

>I can access mysql with a password I was able to create using:
>
>`mysql -uroot -p -h127.0.0.1 --protocol=tcp`
>
>There is nothing in: `/user/local/var/mysql`

The correct path is `/usr/local/var/mysql`. If you actually did `ls /user/local/var/mysql`, you'd definitely get a "no such file or directory" error.

What's the output of `cat /usr/local/var/mysql/$(hostname).err`?

**Reply by websitejunction** [reply]
All great feedback. The was a typo in my path and yes, it's `/usr/local/var/mysql`. 

> The MySQL folks specifically recommend NOT running as root (https://mariadb.com/kb/en/running-mysqld-as-root/), and nothing in the Homebrew setup for mysql requires it. Homebrew also gave you another reason not to do so:

This is my issue. I uninstalled homebrew and resintall and am now seeing:

Name  Status  User         File
```
httpd started root         ~/Library/LaunchAgents/homebrew.mxcl.httpd.plist
mysql stopped root         ~/Library/LaunchAgents/homebrew.mxcl.mysql.plist
php   started willbermudez ~/Library/LaunchAgents/homebrew.mxcl.php.plist
```
All was installed via homebrew and that `root` is killing me.

There is nothing in `/usr/local/var/mysql/` and I don't understand why.

I am able to run` tail -f /opt/homebrew/var/log/httpd/error_log` and get:

```
[Sun Mar 12 19:09:10.771351 2023] [core:notice] [pid 25138] AH00094: Command line: '/opt/homebrew/opt/httpd/bin/httpd -D FOREGROUND'
[Sun Mar 12 19:12:14.197983 2023] [mpm_prefork:notice] [pid 25138] AH00169: caught SIGTERM, shutting down
[Sun Mar 12 19:12:15.268656 2023] [mpm_prefork:notice] [pid 25538] AH00163: Apache/2.4.56 (Unix) configured -- resuming normal operations
[Sun Mar 12 19:12:15.269103 2023] [core:notice] [pid 25538] AH00094: Command line: '/opt/homebrew/opt/httpd/bin/httpd -D FOREGROUND'
[Sun Mar 12 19:15:56.285924 2023] [mpm_prefork:notice] [pid 25538] AH00169: caught SIGTERM, shutting down
[Sun Mar 12 19:15:57.347662 2023] [mpm_prefork:notice] [pid 25978] AH00163: Apache/2.4.56 (Unix) configured -- resuming normal operations
[Sun Mar 12 19:15:57.348050 2023] [core:notice] [pid 25978] AH00094: Command line: '/opt/homebrew/opt/httpd/bin/httpd -D FOREGROUND'
[Sun Mar 12 19:20:49.119799 2023] [mpm_prefork:notice] [pid 25978] AH00169: caught SIGTERM, shutting down
[Sun Mar 12 19:20:50.198111 2023] [mpm_prefork:notice] [pid 26606] AH00163: Apache/2.4.56 (Unix) configured -- resuming normal operations
[Sun Mar 12 19:20:50.198420 2023] [core:notice] [pid 26606] AH00094: Command line: '/opt/homebrew/opt/httpd/bin/httpd -D FOREGROUND'
```

I guess the question is now, why does `httpd `and `mysql` show `root `for the user when I used homebrew to install it?

**Comment by websitejunction** [comment]
All sorted. Thanks.

**Reply by prrraveen** [reply]
what was the issue?

**Reply by fietenoer** [reply]
I have run into the exact same issue and haven't really found a fix yet. It worked flawlessly and suddenly broke without me making any changes. It must have been some update I guess?

For me right now what isn't a good solution but works for now at least is to run `/opt/homebrew/opt/mysql/bin/mysqld_safe` in a terminal and then close that terminal. That leaves the process running in the background. Hopefully I can fix the underlying issue at some point though.
